src/api/spec/controllers/webui/user_controller_spec.rb

```diff
@@ -220,6 +220,38 @@
         expect(user.roles).to match_array old_global_role
       end
     end
+
+    context "when in LDAP mode" do
+      before do
+        login user
+        stub_const('CONFIG', CONFIG.merge({ 'ldap_mode' => :on }))
+      end
+
+      it "users are not allowed to change their accounts" do
+        post :save, params: { user: { login: user.login }, realname: 'another real name', email: 'new_valid@email.es' }
+        expect(controller).to set_flash[:error]
+      end
+
+      it "admins can not change real names of users" do
+        login admin_user
+        old_realname = user.realname
+
+        post :save, params: { user: { login: user.login }, realname: 'new real name', email: 'new_valid@email.es' }
+
+        user.reload
+        expect(user.realname).to eq(old_realname)
+      end
+
+      it "admins can not change email addresses of users" do
```

Comment:
Because the behavior is completely different. An admin user can still edit some parts of a user while a user can edit nothing. Thus I check if the flash was set.